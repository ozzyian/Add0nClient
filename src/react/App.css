.center {
  position: absolute;
  left: 20%;
  margin-left: -50px;
  top: 50%;
  margin-top: -50px;
  width: 70%;
}

.center-spinner {
  position: absolute;
  left: 50%;
  margin-left: -50px;
  top: 50%;
  margin-top: -50px;
}

#container {
  margin-top: 2em;
}
