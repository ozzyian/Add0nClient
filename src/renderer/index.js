// Initial welcome page. Delete the following line to remove it.
const body = document.getElementsByTagName('body')[0];

if (window.matchMedia('(prefers-color-scheme:dark)').matches) {
  body.style.color = 'white';
}

const header = document.createElement('h1');
header.innerText = 'En header';
body.appendChild(header);
