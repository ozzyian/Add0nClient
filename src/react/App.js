import React from 'react';
import AddonTable from './components/AddonTable';
import 'bootstrap/dist/css/bootstrap.css';
import './App.css';
const ipc = require('electron').ipcRenderer;

/**
 *
 */
class App extends React.Component {
  /**
   *
   * @param {*} props
   */
  constructor(props) {
    super(props);
    this.onChange = this.onChange.bind(this);
    this.state = {initiated: false, addons: []};
  }

  /**
   *
   */
  async componentDidMount() {
    const res = await ipc.invoke('app-load');
    if (!res) {
      return;
    } else {
      this.setState({initiated: true, addons: res});
    }
  }
  /**
   *
   * @param {Object} event
   */
  async onChange(event) {
    const path = event.target.files[0].path.split('\\');
    path.pop();
    const addons = await ipc.invoke('init-with-path', path.join('/'));
    this.setState({initiated: true, addons: addons});
  }
  /**
   *
   * @param {*} e
   */

  /**
   * @return {*}
   */
  render() {
    if (this.state.initiated) {
      return (
        <AddonTable
          initiated={this.state.initiated}
          addons={this.state.addons}
        ></AddonTable>
      );
    } else {
      return (
        <div className="custom-file center">
          <input
            type="file"
            className="custom-file-input"
            id="customFile"
            onChange={(e) => this.onChange(e)}
          ></input>
          <label
            className="custom-file-label"
            htmlFor="customFile"
            overflow="false"
          >
            Choose Wow.exe or ClassicWoW.exe
          </label>
        </div>
      );
    }
  }
}

export default App;
