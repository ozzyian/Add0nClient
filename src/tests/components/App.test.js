import React from 'react';
import App from '../../react/App';
import renderer from 'react-test-renderer';

describe('Snapshot test for App.js', () => {
  it('renders correctly when not initialized', () => {
    const tree = renderer.create(<App></App>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when initialized', () => {
    const component = renderer.create(<App></App>);
    component.getInstance().onChange({
      target: {
        files: [{path: 'path'}],
      },
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
});
