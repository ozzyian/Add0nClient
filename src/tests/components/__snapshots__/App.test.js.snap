// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for App.js renders correctly when initialized 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        ID
      </th>
      <th>
        Addon
      </th>
      <th>
        Status
      </th>
      <th>
        Version
      </th>
      <th>
        Author
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;

exports[`Snapshot test for App.js renders correctly when not initialized 1`] = `
<div
  className="custom-file center"
>
  <input
    className="custom-file-input"
    id="customFile"
    onChange={[Function]}
    type="file"
  />
  <label
    className="custom-file-label"
    htmlFor="customFile"
    overflow="false"
  >
    Choose Wow.exe or ClasicWoW.exe
  </label>
</div>
`;
